webapp/views.py

```diff
@@ -72,72 +75,142 @@ def _normalise_resources(self, resources):
                     name=self.CATEGORIES[content]['name']
                 )
+                feed_items['posts'] = {}
+                feed_items['posts'][content] = {}
+                feed_items['posts'][content]['posts'] = posts
+                feed_items['posts'][content]['group_name'] = title
             else:
                 category_id = str(self.GROUPS[topic]['id'])
                 content_id = str(self.CATEGORIES[content]['id'])
                 api_url = (
                     '{api_url}/posts?_embed'
                     '&group={category_id}&categories={content_id}'
+                    '&per_page={per_page}'
+                    '&page={page}'
                 )
                 api_url = api_url.format(
                     api_url=self.API_URL,
                     category_id=category_id,
                     content_id=content_id,
+                    per_page=self.PER_PAGE,
+                    page=page,
+                )
+                posts = get_json_feed_content(api_url)
+                feed_items['pagination'] = self._generate_pagination_queries(
+                    page,
+                    len(posts)
                 )
-                feed_items[topic] = {}
-                feed_items[topic]['items'] = get_json_feed_content(api_url)
-                feed_items[topic]['group_name'] = '{topic} {content}'.format(
+                title = '{topic} {content}'.format(
                     topic=self.GROUPS[topic]['name'],
                     content=self.CATEGORIES[content]['name'],
                 )
+                feed_items['posts'] = {}
+                feed_items['posts'][topic] = {}
+                feed_items['posts'][topic]['posts'] = posts
+                feed_items['posts'][topic]['group_name'] = title
         else:
             for group_name, group in self.GROUPS.items():
                 api_url = (
                     '{api_url}/posts'
-                    '?group={group_id}&{resource_filter}'
+                    '?group={group_id}&{resource_filter}&per_page={per_page}'
+                    '&page={page}'
                 ).format(
                     api_url=self.API_URL,
                     group_id=group['id'],
                     resource_filter=self.RESOURCE_FILTER,
+                    per_page=self.PER_PAGE,
+                    page=page,
                 )
-                feed_items[group_name] = {}
-                feed_items[group_name]['items'] = get_json_feed_content(
-                    api_url,
-                    limit=3,
+                posts = get_json_feed_content(api_url, limit=3)
+                feed_items['pagination'] = self._generate_pagination_queries(
+                    page,
+                    len(posts)
                 )
+                feed_items['posts'] = {}
+                feed_items['posts'][group_name] = {}
+                feed_items['posts'][group_name]['posts'] = posts
                 feed_items[group_name]['group_name'] = group['name']
```

Comment:
I'm getting a KeyError in python from this line of code

Reply:
Fixed